src/urg_node_driver.cpp

```diff
@@ -31,84 +31,97 @@
  * Author: Chad Rockey, Michael Carroll, Mike O'Driscoll
  */
 
-#include "urg_node/urg_node_driver.h"
-
-#include <tf/tf.h>  // tf header for resolving tf prefix
-#include <string>
-#include <diagnostic_msgs/DiagnosticStatus.h>
-#include <urg_node/Status.h>
+#include "urg_node/urg_node_driver.hpp"
 
 namespace urg_node
 {
 
 // Useful typedefs
 typedef diagnostic_updater::FrequencyStatusParam FrequencyStatusParam;
 
-UrgNode::UrgNode(ros::NodeHandle nh, ros::NodeHandle private_nh) :
-  nh_(nh),
-  pnh_(private_nh)
-{
-  initSetup();
-}
-
-UrgNode::UrgNode():
-  nh_(),
-  pnh_("~")
+UrgNode::UrgNode(const std::string & node_name)
+: Node(node_name),
+  diagnostic_updater_(this),
+  error_code_(0),
+  error_count_(0),
+  error_limit_(4),
+  lockout_status_(false),
+  close_diagnostics_(true),
+  close_scan_(true),
+  ip_address_(""),
+  ip_port_(10940),
+  serial_port_("/dev/cu.usbmodem141101"),
+  serial_baud_(115200),
+  calibrate_time_(false),
+  publish_intensity_(false),
+  publish_multiecho_(false),
+  diagnostics_tolerance_(0.05),
+  diagnostics_window_time_(5.0),
+  detailed_status_(false),
+  angle_min_(-3.14),
+  angle_max_(3.14),
+  cluster_(1),
+  skip_(0),
+  default_user_latency_(0.0),
+  laser_frame_id_("laser"),
+  service_yield_(true)
 {
+  (void) synchronize_time_;
```

Comment:
Is this line necessary?

Reply:
I believe this yields a warning of `unused variable` otherwise. I don't know too much about the original use case of this synchronization as for I just "commented it out".

Reply:
I don't think it does cause a warning.  Because this is now a class variable, the compiler will no longer generate the `unused variable` warning (I confirmed this locally).  So I'd just remove this.